cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+
+    @View.nested
+    class advanced(View):  # noqa
+        pre_playbook = BootstrapSelect("preMigrationPlaybook")
+        post_playbook = BootstrapSelect("postMigrationPlaybook")
+        pre_checkbox = Text(locator='.//input[contains(@id, "pre_migration_select_all")]')
+        post_checkbox = Text(locator='.//input[contains(@id, "post_migration_select_all")]')
+
+        @property
+        def is_displayed(self):
+            return self.pre_playbook.is_displayed
+
+        def fill(self, values):
+            self.pre_playbook.wait_displayed("5s")
+            pre_playbook = values.get('pre_playbook')
+            post_playbook = values.get('post_playbook')
+            if pre_playbook:
+                self.pre_playbook.fill(pre_playbook)
+                self.pre_checkbox.click()
+            if post_playbook:
+                self.post_playbook.wait_displayed("5s")
+                self.post_playbook.fill(post_playbook)
+                self.post_checkbox.click()
+            self.after_fill(was_change=True)
+
+        def after_fill(self, was_change):
+            self.parent.next_btn.click()
+
+    @View.nested
+    class schedule(View):  # noqa
+        create = Button("Create")
+        run_migration = RadioGroup('.//div[contains(@id,"migration_plan_choice_radio")]')
+
+        @property
+        def is_displayed(self):
+            return self.run_migration.is_displayed
+
+        def fill(self, values):
+            if values:
+                self.run_migration.select("Start migration immediately")
+            self.after_fill(was_change=True)
+
+        def after_fill(self, was_change):
+            self.create.click()
+
+    @View.nested
+    class results(View):  # noqa
+        msg = Text('.//h3[contains(@id,"migration-plan-results-message")]')
+
+        @property
+        def is_displayed(self):
+            return self.msg.is_displayed
+
+    def after_fill(self, was_change):
+        self.close.click()
+
+    @property
+    def is_displayed(self):
+        return self.title.text == "Create Migration Plan"
+
+
+class MigrationPlanRequestDetailsView(View):
+    migration_request_details_list = MigrationPlanRequestDetailsList("plan-request-details-list")
+    sort_type = SelectorDropdown("id", "sortTypeMenu")
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu")
+    filter_by_status_dropdown = SelectorDropdown("id", "filterCategoryMenu")
+
+    @property
+    def is_displayed(self):
+        return self.migration_request_details_list.is_displayed
+
+    def after_fill(self, was_change):
+        self.browser.send_keys(Keys.ENTER, self.search_box)
+
+    def filter_by_vm_name(self, vm_name):
```

Comment:
Is this really a fill method? Can we accept options for filter_by_dropdown other than 'VM Name'?